### PR TITLE
ref(trace-view): Make root optional for traces

### PR DESCRIPTION
**Pull request by wmak**
- Now that we have orphan traces included in the response, its
  possible that for a trace id we only have orphan traces. Which
  means that we were 204ing in a case where we could've returned a
  valid response
- This also means that we had to remove root always being returned in
  the light trace response, this is because with orphan traces we can no
  longer guarantee that the root is connected to the current
  transaction. This means that the (root)-(this event) link could
  potentially be incorrect during the full trace loading state
- **Note** that in the case of a trace with no root, the full trace may fail to return the current event in the 1st element of the `trace_results`
    - We currently 204 in these cases so still an improvement
    - And we'll be consuming the orphan traces before GA so this will no longer be a problem once we do that.

tldr; root transactions are iffy in traces so only return when we're absolutely sure.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 1]
```diff
@@ -262,16 +255,24 @@ def serialize(
...
+                    trace_results.append(
+                        self.serialize_event(
+                            root,
+                            None,
+                            0,
+                        )
                     )
-                )
+            else:
+                is_root_child = None
```
Comment (Zylphrex) [inline src/sentry/api/endpoints/organization_events_trace.py:268]:
```suggestion
                is_root_child = False
```

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 2]
```diff
@@ -262,16 +255,24 @@ def serialize(
...
 
-            # The current event should be the last item in the trace_results
-            current_event = trace_results[-1]
+            current_event = self.serialize_event(
+                snuba_event,
+                root["id"] if is_root_child else None,
+                1 if is_root_child else 0 if root and root["id"] == snuba_event["id"] else None,
```
Comment (Zylphrex) [inline src/sentry/api/endpoints/organization_events_trace.py:273]: Can we refactor this out of this one-liner?

Comment (dashed) [reply src/sentry/api/endpoints/organization_events_trace.py:273]: Very Pythonic.

**Review by Zylphrex — APPROVED** [review]
